src/StatusQ/Controls/StatusInput.qml

```diff
@@ -0,0 +1,85 @@
+import QtQuick 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+Rectangle {
+    id: root
+
+    implicitWidth: 480
+    height: (label.visible ? 
+                  label.anchors.topMargin +
+                  label.height :
+              charLimitLabel.visible ?
+                  charLimitLabel.anchors.topMargin +
+                  charLimitLabel.height :
+              0) +
+            statusBaseInput.anchors.topMargin +
+            statusBaseInput.height + 
+            (errorMessage.visible ? 
+                  errorMessage.anchors.topMargin +
+                  errorMessage.height :
+                  0) + 8
+
+    color: "transparent"
+    property alias input: statusBaseInput
+    property string label: ""
+    property int charLimit: 0
+    property string errorMessage: ""
+
+    StatusBaseText {
+        id: label
+        height: visible ? implicitHeight : 0
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.topMargin: visible ? 8 : 0
+        anchors.leftMargin: 16
+        visible: !!root.label
+
+        text: root.label
```

Comment:
is elide enabled by default? shall we set the max width for the text then?

Reply:
Yea wouldn't hurt! Although, in practice I think chances are very low we'd exceed the width

Reply:
In my experience it started to overflow when your app gets translated to other languages... translators (humans) will never consider allowed boundaries.

Reply:
Yeap, very good argument. Updating this.